rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -210,7 +210,16 @@ rclcpp::remove_ros_arguments(int argc, char const * const argv[])
 
   ret = rcl_parse_arguments(argc, argv, alloc, &parsed_args);
   if (RCL_RET_OK != ret) {
-    exceptions::throw_from_rcl_error(ret, "Failed to parse arguments");
+    // Not using throw_from_rcl_error, because we may need to append deallocation failures.
+    exceptions::RCLErrorBase base_exec(ret, rcl_get_error_state());
+    rcl_reset_error();
+    if (RCL_RET_OK != rcl_arguments_fini(&parsed_args)) {
```

Comment:
It looks like `rcl_parse_arguments` [cleans up after itself](https://github.com/ros2/rcl/blob/d41c923927850274f63aee1bde565d3a8266a7bd/rcl/src/rcl/arguments.c#L303-L311) when something goes wrong. I would expect this call to `rcl_arguments_fini` to always fail [here](https://github.com/ros2/rcl/blob/d41c923927850274f63aee1bde565d3a8266a7bd/rcl/src/rcl/arguments.c#L432).